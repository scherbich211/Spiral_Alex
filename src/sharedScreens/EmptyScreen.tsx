import React from 'react';
import {StatusBar, StyleSheet, View, ViewStyle} from 'react-native';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
import {useTheme} from 'react-native-paper';
import Icon from 'react-native-vector-icons/Ionicons';
import TabHeader from '../components/Header/TabHeader';
import Typography from '../components/Typography';
import {SIZES} from '../theme';

export interface IEmptyStyles {
	tapBar: ViewStyle;
	headerText: ViewStyle;
	mainPart: ViewStyle;
}

interface IProps {
	navigateBack: () => void;
	title: string;
	screenName: string;
}

const EmptyScreen: React.FC<IProps> = props => {
	const theme = useTheme();
	const {title} = props;
	const styles = useStyles(title);

	return (
		<View style={styles.tapBar}>
			<StatusBar barStyle="dark-content" />
			<TabHeader
				beforeText={
					<View>
						<Icon.Button
							name="arrow-back"
							size={25}
							backgroundColor={theme.colors.primary}
							onPress={() => props.navigateBack()}
						/>
					</View>
				}
				headerText={props.title}
				headerStyle={styles.headerText}
			/>
			<KeyboardAwareScrollView>
				<View style={styles.mainPart}>
					<Typography>{props.screenName}</Typography>
				</View>
			</KeyboardAwareScrollView>
		</View>
	);
};

const useStyles = StyleSheet.create(
	(title: string): IEmptyStyles => ({
		tapBar: {
			flex: 1,
		},
		headerText: {
			marginLeft: title.length > 6 ? SIZES.width * 0.25 : SIZES.width * 0.3,
		},
		mainPart: {flex: 1, justifyContent: 'center', alignItems: 'center', height: SIZES.height * 0.8},
	}),
);

export default EmptyScreen;
