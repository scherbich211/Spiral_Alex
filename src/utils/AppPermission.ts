import {
	AndroidPermission,
	check,
	IOSPermission,
	PERMISSIONS,
	request,
	RESULTS,
	WindowsPermission,
} from 'react-native-permissions';

import {Platform} from 'react-native';

interface Camera {
	ios?: string;
	android?: string;
	windows?: string;
	macos?: string;
	web?: string;
}

interface REQUEST_PERMISSION_TYPE {
	[key: string]: Camera;
}

const PLATFORM_CAMERA_PERMISSIONS = {
	ios: PERMISSIONS.IOS.CAMERA,
};

const PLATFORM_PHOTO_PERMISSIONS = {
	ios: PERMISSIONS.IOS.PHOTO_LIBRARY,
};

const REQUEST_PERMISSION_TYPE: REQUEST_PERMISSION_TYPE = {
	camera: PLATFORM_CAMERA_PERMISSIONS,
	photo: PLATFORM_PHOTO_PERMISSIONS,
};

const PERMISSION_TYPE = {
	camera: 'camera',
	photo: 'photo',
};

class AppPermission {
	checkPermission = async (type: string): Promise<boolean> => {
		console.log('AppPermission checkPermission type', type);
		const permissions = REQUEST_PERMISSION_TYPE[type][Platform.OS];
		console.log('AppPermission checkPermission permissions', permissions);

		if (!permissions) {
			return true;
		}
		try {
			const result = await check(permissions as AndroidPermission | IOSPermission | WindowsPermission);
			console.log('AppPermission checkPermission result', result);

			if (result === RESULTS.GRANTED) {
				console.log('ENDS');
				return true;
			}
			return this.requestPermission(permissions);
		} catch (error) {
			console.log('AppPermission checkPermission error', error);

			return false;
		}
	};

	requestPermission = async (permissions: string): Promise<boolean> => {
		console.log('AppPermission requestPermission permissions', permissions);
		try {
			const result = await request(permissions as AndroidPermission | IOSPermission | WindowsPermission);
			console.log('AppPermission requestPermission result', result);

			return result === RESULTS.GRANTED;
		} catch (error) {
			console.log('AppPermission requestPermission error', error);

			return false;
		}
	};
}

// const componentDidMount = () => {
// 	Permission.checkPermission(PERMISSION_TYPE.camera);
// };

const Permission = new AppPermission();
export {Permission, PERMISSION_TYPE};
