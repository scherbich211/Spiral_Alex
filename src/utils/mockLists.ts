import {RootStackParamList} from '../types';

export interface IListData {
	id: number;
	title: keyof RootStackParamList;
	subtitle: string;
	icon: string;
	dollars: string;
	cents: string;
}

export const dollars = [1500, 5000, 500];
export const cents = [20, 20, 40];

export const list: IListData[] = [
	{
		id: 1,
		title: 'Checking',
		subtitle: 'Main account (...0353)',
		icon: 'arrow-forward',
		dollars: `$${dollars[0]}`,
		cents: `.${cents[0]}`,
	},
	{
		id: 2,
		title: 'Savings',
		subtitle: 'Buy a house (...4044)',
		icon: 'arrow-forward',
		dollars: `$${dollars[1]}`,
		cents: `.${cents[1]}`,
	},
	{
		id: 3,
		title: 'Goodness',
		subtitle: 'Cash Rewards',
		icon: 'arrow-forward',
		dollars: `$${dollars[2]}`,
		cents: `.${cents[2]}`,
	},
];

export const dollars11 = [63, 17, 1200, 320];
export const cents11 = [95, 75, 50, 73];

export const list11 = [
	{
		id: 1,
		title: 'Target',
		subtitle: 'Closter NJ | Debit card',
		dollars: `$${dollars11[0]}`,
		cents: `.${cents11[0]}`,
	},
	{
		id: 2,
		title: 'AplPay 7-Eleven',
		subtitle: 'Cresskill NJ | iPhone',
		dollars: `$${dollars11[1]}`,
		cents: `.${cents11[1]}`,
	},
	{
		id: 3,
		title: 'Facebook inc',
		subtitle: 'Pay day! Yay!',
		dollars: `$${dollars11[2]}`,
		cents: `.${cents11[2]}`,
	},
	{
		id: 4,
		title: 'Lencrafters',
		subtitle: 'Paramus NJ | Debit card',
		dollars: `$${dollars11[3]}`,
		cents: `.${cents11[3]}`,
	},
];

export const dollars10 = [10000, 12, 236, 320];
export const cents10 = [0, 2, 52, 73];

export const list10 = [
	{
		id: 1,
		title: 'Transfer from savings',
		subtitle: 'Buy a house (...4044)',
		dollars: `$${dollars10[0]}`,
		cents: `.${cents10[0]}`,
	},
	{
		id: 2,
		title: 'Starbucks',
		subtitle: 'Closter NJ | Debit card',
		dollars: `$${dollars10[1]}`,
		cents: `.${cents10[1]}`,
	},
	{
		id: 3,
		title: 'Stop and Shop',
		subtitle: 'Closter NJ | Debit card',
		dollars: `$${dollars10[2]}`,
		cents: `.${cents10[2]}`,
	},
	{
		id: 4,
		title: 'Lencrafters',
		subtitle: 'Paramus NJ | Debit card',
		dollars: `$${dollars10[3]}`,
		cents: `.${cents10[3]}`,
	},
];

export const listSavings = [
	{
		id: 1,
		title: 'End day balance - Jul 11',
		dollars: '$10,000',
	},
	{
		id: 2,
		title: 'Deposit',
		subtitle: 'Jul 11',
		dollars: '$2,000',
		cents: '.00',
	},
	{
		id: 3,
		title: 'Deposit',
		subtitle: 'Jul 11',
		dollars: '$2,000',
		cents: '.00',
	},
	{
		id: 4,
		title: 'Deposit',
		subtitle: 'Jul 11',
		dollars: '$2,000',
		cents: '.00',
	},
	{
		id: 5,
		title: 'Deposit',
		subtitle: 'Jul 11',
		dollars: '$2,000',
		cents: '.00',
	},
	{
		id: 6,
		title: 'Deposit',
		subtitle: 'Jul 11',
		dollars: '$2,000',
		cents: '.00',
	},
	{
		id: 7,
		title: 'End day balance - Jul 10',
		dollars: '$3,000',
		cents: '.00',
	},
	{
		id: 8,
		title: 'Deposit',
		subtitle: 'Jul 10',
		dollars: '$2,000',
		cents: '.00',
	},
	{
		id: 9,
		title: 'Deposit',
		subtitle: 'Jul 10',
		dollars: '$500',
		cents: '.00',
	},
	{
		id: 10,
		title: 'Deposit',
		subtitle: 'Jul 10',
		dollars: '$500',
		cents: '.00',
	},
];
