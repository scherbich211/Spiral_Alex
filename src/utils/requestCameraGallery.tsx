import {Platform} from 'react-native';
import {ImageLibraryOptions, launchCamera, launchImageLibrary} from 'react-native-image-picker';
import {Permission, PERMISSION_TYPE} from './requestAllPermissions';
import {requestCameraPermission, requestExternalWritePermission} from './requestPermissions';

// -------------takePhoto------------------------------------------------------------
export const captureImage = async (
	type: ImageLibraryOptions['mediaType'],
	changeImage: (url: string) => void,
): Promise<void> => {
	const options: ImageLibraryOptions = {
		mediaType: type,
		maxWidth: 600,
		maxHeight: 1100,
		quality: 1,
	};
	if (Platform.OS === 'android') {
		console.log('android');
		const isCameraPermitted: boolean = await requestCameraPermission();
		const isStoragePermitted: boolean = await requestExternalWritePermission();
		if (isCameraPermitted && isStoragePermitted) {
			launchCamera(options, response => {
				console.log('Response = ', response);
				if (response.didCancel) {
					console.warn('User cancelled camera picker');
					return;
				} else if (response.errorCode === 'camera_unavailable') {
					console.warn('Camera not available on device');
					return;
				} else if (response.errorCode === 'permission') {
					console.warn('Permission not satisfied');
					return;
				} else if (response.errorCode === 'others') {
					console.warn(response.errorMessage);
					return;
				}
				const size: number | undefined = response.assets?.find(i => i.fileSize)?.fileSize;
				const duration = response.assets?.find(i => i.duration)?.duration;

				if ((size && size.toString().length > 0) || duration) {
					if (response.assets) {
						const [asset] = response.assets;
						if (asset) {
							if (asset.uri) {
								changeImage(asset.uri);
							}
						} else {
							console.warn('no asset available');
						}
					} else {
						console.warn('no assets available');
					}
				}
			});
		}
	} else {
		const componentDidMount = (): void => {
			Permission.checkPermission(PERMISSION_TYPE.camera);
		};
		// let isCameraPermitted = requestCameraPermissionIos();
		const isCameraPermitted: void = componentDidMount();
		const isStoragePermitted: boolean = await requestExternalWritePermission();
		if (isCameraPermitted != null && isStoragePermitted != null) {
			launchCamera(options, (response): void => {
				console.log('Response = ', response);
				if (response.didCancel) {
					console.warn('User cancelled camera picker');
					return;
				} else if (response.errorCode === 'camera_unavailable') {
					console.warn('Camera not available on device');
					return;
				} else if (response.errorCode === 'permission') {
					console.warn('Permission not satisfied');
					return;
				} else if (response.errorCode === 'others') {
					console.warn(response.errorMessage);
					return;
				}
				const size: number | undefined = response.assets?.find(i => i.fileSize)?.fileSize;
				const duration = response.assets?.find(i => i.duration)?.duration;

				if ((size && size.toString().length > 0) || duration) {
					if (response.assets) {
						const [asset] = response.assets;
						if (asset) {
							if (asset.uri) {
								// setData({
								// 	...data,
								// 	filePath: asset.uri,
								// });
							}
						} else {
							console.warn('no asset available');
						}
					} else {
						console.warn('no assets available');
					}
				}
			});
		}
	}
};

// -------------chooseFile------------------------------------------------------------
export const chooseFile = (type: ImageLibraryOptions['mediaType'], changeImage: (url: string) => void) => {
	const options: ImageLibraryOptions = {
		mediaType: type,
		maxWidth: 600,
		maxHeight: 1100,
		quality: 1,
	};
	launchImageLibrary(options, response => {
		if (response.didCancel) {
			console.warn('User cancelled camera picker');
			return;
		} else if (response.errorCode === 'camera_unavailable') {
			console.warn('Camera not available on device');
			return;
		} else if (response.errorCode === 'permission') {
			console.warn('Permission not satisfied');
			return;
		} else if (response.errorCode === 'others') {
			console.warn(response.errorMessage);
			return;
		}
		const size: number | undefined = response.assets?.find(i => i.fileSize)?.fileSize;
		if (size && size.toString().length > 0) {
			if (response.assets) {
				const [asset] = response.assets;
				if (asset) {
					if (asset.uri) {
						changeImage(asset.uri);
					}
				} else {
					console.warn('no asset available');
				}
			} else {
				console.warn('no assets available');
			}
		}
	});
};
