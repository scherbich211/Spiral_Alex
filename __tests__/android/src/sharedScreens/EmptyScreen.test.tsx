import { NavigationContainer } from "@react-navigation/native";
import { cleanup, render, waitFor } from "@testing-library/react-native";
import React from "react";
import DrawerNavigator from "../../../../src/screens/HomeScreens/Drawer";
import EmptyScreen from "../../../../src/sharedScreens/EmptyScreen";
import { navigate } from "../../../../src/utils/navigateRef";


jest.mock('react-native', function () {
	const reactNative = jest.requireActual('react-native');
	jest.spyOn(reactNative.Platform, 'select').mockImplementation(() => 'android');
	reactNative.Platform.OS = 'android';
	return reactNative;
});

describe('Empty Screen', () => {
	afterEach(cleanup);
	const props = {
		title: 'Accounts',
		navigateBack: ()=>navigate('Home'),
		screenName: 'Accounts',
	  };

	it('Render', async () => {
		const screen = await waitFor(() =>
			render(<EmptyScreen {...props}/>),
		);
		expect(screen).toBeDefined();
		expect(screen).toMatchSnapshot();
		screen.unmount();
	});

	it('Render & screenName', async () => {
		const screen = await waitFor(() =>
			render(<EmptyScreen {...props}/>),
		);
		const screenName = await screen.getByTestId('test-screenName')
		expect(screenName).toBeDefined();
		expect(screenName).toHaveTextContent('Accounts')
		expect(screen).toMatchSnapshot();
		screen.unmount();
	});

	it('Render & title', async () => {
		const screen = await waitFor(() =>
			render(<EmptyScreen {...props}/>),
		);
		const title = await screen.getByTestId('header')
		expect(title).toBeDefined();
		expect(screen).toMatchSnapshot();
		screen.unmount();
	});

	// it('Render & navigate', async () => {
	// 	const screen = await waitFor(() =>
	// 		render(
	// 			<NavigationContainer>
	// 				<DrawerNavigator />
	// 			</NavigationContainer>
	// 		),
	// 	);
	// 	const navigate = await screen.getByTestId('navigate-empty')
	// 	expect(navigate).toBeTruthy();
	// 	expect(screen).toMatchSnapshot();
	// 	screen.unmount();
	// });
});
