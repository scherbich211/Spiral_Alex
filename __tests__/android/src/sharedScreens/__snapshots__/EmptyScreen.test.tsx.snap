// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Screen Render & screenName 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 2,
          "zIndex": 2,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": "android",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          testID="header"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#6200ee",
                  "borderRadius": 5,
                },
              ]
            }
            testID="navigate-empty"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "padding": 8,
                  },
                  Object {
                    "backgroundColor": "#6200ee",
                    "borderRadius": 5,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 25,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "fontFamily": "android",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "white",
                "fontFamily": "Roboto-Regular",
                "fontSize": 17,
                "fontWeight": "700",
                "letterSpacing": 0.8,
                "marginLeft": 187.5,
              },
            ]
          }
          testID="undefined"
        >
          Accounts
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 1067.2,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "#000000",
                "fontFamily": "RobotoCondensed-Regular",
                "fontSize": 14,
                "fontWeight": "normal",
              },
            ]
          }
          testID="test-screenName"
        >
          Accounts
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Empty Screen Render & title 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 2,
          "zIndex": 2,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": "android",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          testID="header"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#6200ee",
                  "borderRadius": 5,
                },
              ]
            }
            testID="navigate-empty"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "padding": 8,
                  },
                  Object {
                    "backgroundColor": "#6200ee",
                    "borderRadius": 5,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 25,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "fontFamily": "android",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "white",
                "fontFamily": "Roboto-Regular",
                "fontSize": 17,
                "fontWeight": "700",
                "letterSpacing": 0.8,
                "marginLeft": 187.5,
              },
            ]
          }
          testID="undefined"
        >
          Accounts
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 1067.2,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "#000000",
                "fontFamily": "RobotoCondensed-Regular",
                "fontSize": 14,
                "fontWeight": "normal",
              },
            ]
          }
          testID="test-screenName"
        >
          Accounts
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Empty Screen Render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 2,
          "zIndex": 2,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": "android",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          testID="header"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#6200ee",
                  "borderRadius": 5,
                },
              ]
            }
            testID="navigate-empty"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "padding": 8,
                  },
                  Object {
                    "backgroundColor": "#6200ee",
                    "borderRadius": 5,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 25,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "fontFamily": "android",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "white",
                "fontFamily": "Roboto-Regular",
                "fontSize": 17,
                "fontWeight": "700",
                "letterSpacing": 0.8,
                "marginLeft": 187.5,
              },
            ]
          }
          testID="undefined"
        >
          Accounts
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 1067.2,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "#000000",
                "fontFamily": "RobotoCondensed-Regular",
                "fontSize": 14,
                "fontWeight": "normal",
              },
            ]
          }
          testID="test-screenName"
        >
          Accounts
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
